packages/@aws-cdk/aws-eks/lib/kubectl-provider.ts

```diff
@@ -97,6 +97,8 @@ export class KubectlProvider extends NestedStack {
 
     const provider = new cr.Provider(this, 'Provider', {
       onEventHandler: handler,
+      vpc: cluster.kubectlPrivateSubnets ? cluster.vpc : undefined,
```

Comment:
This isn't related to the `placeClusterHandlerInVpc` property. But seems like we should have done this when we introduced private endpoints, consolidating all the related functions into the same network configuration.